internal/layer2/announcer.go

```diff
@@ -217,29 +217,31 @@ func (a *Announce) shouldAnnounce(ip net.IP) dropReason {
 }
 
 // SetBalancer adds ip to the set of announced addresses.
-func (a *Announce) SetBalancer(name string, ip net.IP) {
-	// Call doSpam at the end of the function without holding the lock
-	defer a.doSpam(ip)
-	a.Lock()
-	defer a.Unlock()
-
-	// Kubernetes may inform us that we should advertise this address multiple
-	// times, so just no-op any subsequent requests.
-	if _, ok := a.ips[name]; ok {
-		return
-	}
-	a.ips[name] = ip
+func (a *Announce) SetBalancer(name string, ips []net.IP) {
+	for _, ip := range ips {
+		// Call doSpam at the end of the function without holding the lock
+		defer a.doSpam(ip)
+		a.Lock()
+		defer a.Unlock()
```

Comment:
Defer is executed when the function exits, so in case of two ips it will deadlock on the second a.Lock.
I think it will be simpler to keep this function as it is and call it twice from outside.

Reply:
ah sorry my bad I moved the loop to the caller that makes prefect sense